engine/src/main/java/org/destinationsol/SolApplication.java

```diff
@@ -59,6 +59,9 @@
 
     private UiDrawer uiDrawer;
 
+    private FactionDisplay factionDisplay;
+    private FactionInfo factionInfo;
```

Comment:
Why do you have this as field? Everytime you use this, you first assign it a new value; it might as well be just a local variable.

Reply:
It's instanced when the game is started, and the draw method then calls a method from that instance

Reply:
Yeah, ik. GH makes it a bit hard to specify if you are reviewing only one line or several at once.